features/step_definitions/migration/migration_steps.rb

```diff
@@ -0,0 +1,128 @@
+Given(/^At least (\d+) Xen Hypervisors are available$/) do |arg1|
+  # get hypervisors and count Xen's ones - for migration at least 2
+  hashypervisor_xen = 0
+  @hypervisors = control_node.openstack.hypervisor.list
+  expect(@hypervisors).not_to be_empty
+
+  @hypervisors.each do |h|
+    hypervisor = control_node.openstack.hypervisor.show(h.id)
+    hashypervisor_xen = (hashypervisor_xen + 1) if hypervisor.hypervisor_type == "Xen"
+  end
+  expect(hashypervisor_xen).to be >= arg1.to_i
+end
+
+Given(/^At least (\d+) KVM Hypervisors are available$/) do |arg1|
+  # get hypervisors and count KVM's ones - for migration at least 2
+  hashypervisor_kvm = 0
+  @hypervisors = control_node.openstack.hypervisor.list
+  expect(@hypervisors).not_to be_empty
+
+  @hypervisors.each do |h|
+    hypervisor = control_node.openstack.hypervisor.show(h.id)
+    hashypervisor_kvm = (hashypervisor_kvm + 1) if hypervisor.hypervisor_type == "QEMU"
+  end
+  expect(hashypervisor_kvm).to be >= arg1.to_i
+end
+
+When(/^I create an KVM instance$/) do
+  def delete_old_instances(server_id)
+    puts("old instance #{server_id } found - deleting")
+    control_node.exec!("openstack server delete #{server_id}")
+  end
```

Comment:
Would be better to move this method into step helpers https://github.com/SUSE-Cloud/cct/blob/master/features/support/step_helpers.rb , then you don't need to define it again in the context of the next step.